code/datums/radiation_wave.dm

```diff
@@ -110,10 +112,16 @@
 			/obj/item/implant,
 			/obj/singularity
 			))
-		if(!can_contaminate || blacklisted[thing.type])
+		if(!can_contaminate || !can_contam || blacklisted[thing.type])
+			continue
+		if(CHECK_BITFIELD(thing.rad_flags, RAD_NO_CONTAMINATE) || SEND_SIGNAL(thing, COMSIG_ATOM_RAD_CONTAMINATING, strength) & COMPONENT_BLOCK_CONTAMINATION)
 			continue
-		if(prob(contamination_chance)) // Only stronk rads get to have little baby rads
-			if(SEND_SIGNAL(thing, COMSIG_ATOM_RAD_CONTAMINATING, strength) & COMPONENT_BLOCK_CONTAMINATION)
-				continue
-			var/rad_strength = (strength-RAD_MINIMUM_CONTAMINATION) * RAD_CONTAMINATION_STR_COEFFICIENT
+		contam_atoms += thing
+	var/did_contam = 0
+	if(length(contam_atoms) && prob(50+steps*10)) // the prob lets it carry a bit further
+		var/rad_strength = min((strength-RAD_MINIMUM_CONTAMINATION) * RAD_CONTAMINATION_STR_COEFFICIENT,remaining_contam)/contam_atoms.len
+		for(var/k in 1 to contam_atoms.len)
```

Comment:
As I said in the other review this can and should all be done in one loop

Reply:
...can you explain how? I'm building contam_atoms, then using its length to calculate the contamination strength, then applying contamination of that strength to all the items, and I can't damn well do step 3 before 2 or 2 before 1

Reply:
You don't need to divide it up equally between all the possible objects to contaminate and honestly it'll result in some weird behavior.

Reply:
If I don't do this, it'll be allowed to grow infinitely forever again, unless I'm doing my math very wrong.

Reply:
you can still reduce the total contamination each time you apply it and stop contaminating things when you run out